apps/passport-client/src/dispatch.ts

```diff
@@ -842,15 +842,29 @@ async function doSync(
         serverStorageHash: upRes.value.storageHash
       };
     } else {
-      const res: Partial<AppState> = {
-        completedFirstSync: true
-      };
+      // Upload failed.  Update AppState if necessary, but not unnecessarily.
+      // AppState updates will trigger another upload attempt.
+      const needExtraDownload = upRes.error.name === "Conflict";
+      if (
+        state.completedFirstSync &&
+        (!needExtraDownload || state.extraDownloadRequested)
+      ) {
+        return undefined;
+      }
 
+      const updates: Partial<AppState> = {};
+      if (!state.completedFirstSync) {
+        // We completed a first attempt at sync, even if it failed.
+        updates.completedFirstSync = true;
+      }
+      if (needExtraDownload && !state.extraDownloadRequested) {
+        updates.extraDownloadRequested = true;
+      }
       if (upRes.error.name === "ValidationError") {
-        res.userInvalid = true;
+        updates.userInvalid = true;
```

Comment:
In case you're not already working on this, note that after the branch merge, the early-return above causes this line to be skipped when it shouldn't be.

Reply:
The setting of userInvalid needs conditionality in the same way as the setting of the other variables, to avoid a failed upload triggering its own immediate retry.